### ISSUE TITLE
Karma-coverage hangs when run via gulp

### ISSUE THREAD

**Issue by sspansel** [sspansel/Angular-Kickstart#12]
This isn't a huge deal, because when I run the unit tests via "karma start", it runs fine. It actually runs fine via "gulp test" also, but it hangs and I have to kill it via ctrl-c. 

This seems to maybe be related to [this issue on the karma-coverage project](https://github.com/karma-runner/karma-coverage/issues/209).

**Comment by sspansel** [comment]
I don't think it has to do with the issue I previously linked. It seems to be a promise (in homeController.setColumnData or homeService.getColumnData) that isn't getting resolved.

**Comment by sspansel** [comment]
Resolved by changing from gulp 3 to gulp 4.

It was some sort of timing issue, wherein a promise got resolved slightly after the task had completed. It was a known issue in gulp 3 that has been fixed in gulp 4. :)

**Comment by sspansel** [comment]
I'd thought (see previous comment) that upgrading to gulp 4 resolved this issue. But now it has returned.

I still think it's some sort of promise resolution timing issue. But I can also still work around it by doing "karma start" instead of "gulp test". So I'm leaving it as low priority for now.

**Comment by sspansel** [comment]
More info. Gulp does eventually conclude, but only after about 30 seconds:

`time gulp test

[09:38:13] Using gulpfile path/to/gulpfile.js
[09:38:13] Starting 'test'...
<snip>
[09:38:14] Finished 'test' after 698 ms

real	0m31.259s
user	0m1.268s
sys	0m0.116s
`
